common/djangoapps/student/views.py

```diff
@@ -523,15 +538,14 @@ def dashboard(request):
     # Check to see if the student has recently enrolled in a course.
     # If so, display a notification message confirming the enrollment.
     enrollment_message = _create_recent_enrollment_message(
-        course_enrollment_pairs, course_modes_by_course
+        course_enrollments, course_modes_by_course
     )
 
-    # Retrieve the course modes for each course
-    enrolled_courses_dict = {}
-    for course, __ in course_enrollment_pairs:
-        enrolled_courses_dict[unicode(course.id)] = course
-
-    credit_messages = _create_credit_availability_message(enrolled_courses_dict, user)
+    enrolled_course_names_dict = {
+        unicode(enrollment.course_id): enrollment.course_overview.display_name
+        for enrollment in course_enrollments
+    }
+    credit_messages = _create_credit_availability_message(enrolled_course_names_dict, user)
```

Comment:
for future code maintainability, consider passing in enrolled_course_overviews_dict instead of just the names of the courses.